test/testdata/rewriter/minitest_tables.rb

```diff
@@ -0,0 +1,53 @@
+# typed: true
+
+class A
+  def foo; end
+end
+class B < A
+end
+
+class MyTest
+  def outside_method
+  end
+
+  def self.test_each(arg, &blk); end
+  def self.it(name, &blk); end
+
+  test_each [A.new, B.new] do |value|
```

Comment:
Array literals will suffer the same fate that local variables currently suffer from: the thing to compute any given element of the array literal might reference methods on `T.class_of(MyTest)`, but by rewriting the array literal with to `[...].compact.first` inside the `def test_...` method, those method calls will be on the wrong scope.

Can you add a test for this?

Reply:
```ruby
def self.outside_self_method; end
test_each [outside_self_method] do |value|; end
```

Reply:
bump on adding a test for this, even if it's just to say that it errors or is untyped.